### ISSUE TITLE
unrelate: true + noUnrelate does not work as expected in upsertGraph

### ISSUE THREAD

**Issue by unhawkable** [Vincit/objection.js#761]
Hi,
It seems that if I specify unrelate: true + noUnrelate: [ array of relation names ] then it does not work as one might have expected.
I have a model with HasManyRelation (without join table) and I'd like to have the records deleted from that relation if they are missing from a graph. All other missing records from other relations should just become unrelated.

But, nothing happens to these records. They are neither unrelated nor deleted.
To make it work, I need to explicitly specify relation names in unrelate: [] which i want to be unrelated and remove noUnrelate option.

**Comment by koskimas** [comment]
graphUpsert is such a complex beast that unfortunately I need a reproduction before I start to investigate this. There are 1000 things you may be doing differently than what I'd expect. There's a `reproduction-template.js` file in the repo root if you prefer to use that.

**Comment by koskimas** [comment]
There's a at least one test for that specific case [here](https://github.com/Vincit/objection.js/blob/a758ae8bdb5fc21a211e99ee8d20d0b4e309180f/tests/integration/upsertGraph.js#L845). You may have found a corner case bug though.

**Comment by koskimas** [comment]
@unhawkable Did you solve this? Can I close this issue? If not, please answer something so that I know that you are working on a reproduction.

**Comment by unhawkable** [comment]
Thank you for super-fast response. I will work on the reproduction now.

**Comment by unhawkable** [comment]
Please find the reproduction file attached:
[reproduction-template.zip](https://github.com/Vincit/objection.js/files/1674071/reproduction-template.zip)

It seems to fail under sqlite as well. The record from "noUnrelate" relation is not being deleted.

**Comment by unhawkable** [comment]
So noUnrelate does not necessarily mean - delete that record instead, right? But could we maybe have the records deleted with some other combination of options?

**Comment by koskimas** [comment]
Yes, noUnrelate doesn't revert the functionality back to deleting. Deleting is the default. Unfortunately there currently is no other way but to explicitly list all relations except the one you want to delete in `unrelate`.
